framework/src/controller/application_state.js

```diff
@@ -78,19 +78,18 @@ class ApplicationState {
 	}
 
 	/**
-	 * Updates broadhash and height values.
+	 * Updates prevotedConfirmedUptoHeight and height values.
 	 *
-	 * @param {broadhash, height} parameters - broadhash and height to update
+	 * @param {prevotedConfirmedUptoHeight, height} parameters - prevotedConfirmedUptoHeight and height to update
 	 *
 	 * @returns {Promise.<boolean, Error>}
 	 * @throws assert.AssertionError
 	 */
-	async update({ broadhash, height }) {
-		assert(broadhash, 'broadhash is required to update application state.');
+	async update({ height, prevotedConfirmedUptoHeight = 0 }) {
 		assert(height, 'height is required to update application state.');
 		try {
 			const newState = this.state;
-			newState.broadhash = broadhash;
+			newState.prevotedConfirmedUptoHeight = prevotedConfirmedUptoHeight;
```

Comment:
Concern about this line:
The default value for `prevotedConfirmedUpToHeight` is `0`.

What if we don't want to update the `prevotedConfirmedUpToHeight`, then it will set the value to `0`. Or is this case not possible atm?

I was thinking about adding a check that the new value must be bigger than the current `prevotedConfirmed`, but when kicking in Block Sync Mechanism, the new value can be lower than the current one.

Or remove the default value for `prevotedConfirmedUpToHeight` as you already set the initial state to `0` in `line 66`.

Reply:
That's a good point @michielmulders and I agree with you. The concrete implementation of this function was too tied to the concept of broadhash (you couldn't update another property of the state without not sending the broadhash ) and now it remains the same behavior for `prevotedConfirmedUpToHeight`.

At the moment there is only one place where the application state is updated but if someone else wants to update the application state from another part of the code this will result in wrongly setting the attribute to `0`.

> I was thinking about adding a check that the new value must be bigger than the current prevotedConfirmed, but when kicking in Block Sync Mechanism, the new value can be lower than the current one.

This shouldn't be part of the logic of the application state. I like better your idea of removing the default of `0`, but then if someone else updates the application state without passing the attribute it will result in a null value.

My opinion is that the application state `update` function should be generic, have schema validation (to not pass wrong formats or different attributes), and it should not require certain attributes to be passed explicitly. But this is the scope of another issue.

Reply:
I would also consider getters and setter for having an strict control over each shared property exposed by application state.

Reply:
Getters and setters I agree with! 👍 
And @limiaspasdaniel , I would remove the default value for the input parameters in the update function, but leave the initialisation of the state to `0` so a `null` value is not possible.